### DISCUSSION TITLE
asyncio.wait_until()

### DISCUSSION THREAD

**Discussion by mrracoonman** [Rapptz/discord.py#6560] [Q&A]
This is a short example of my code.

    "imports and all that"

    @bot.command(aliases=["p"])
    async def play(ctx, *, arg=None):
        guild = ctx.guild
        voice_client: discord.VoiceClient = discord.utils.get(bot.voice_clients, guild=guild)
        channel = ctx.author.voice.channel
        print(f"Bot has connected to {channel}.")

        if arg == "__song__":
            audio_source = discord.FFmpegPCMAudio(
               __path to mp3 file__')
                print(f"__song__ is playing.")
            if not voice_client.is_playing():
                voice_client.play(audio_source, after=lambda e: print(f'__song__ has finished playing.'))
                await ctx.send("Starting now.")

And I've done a little research, but I wasn't able to find a way to use asyncio.wait_until() to wait until the voice_client is finished playing to do another voice_client.play(). Am I using it in the wrong scenario and if so, is there something I can do to suceed in what I'm wanting?

python 3.8

**Comment by lgaan** [comment]
As you can see, `voice_client.play` has an `after` parameter. This can be used to play another song upon the current `.play` call finishing.

**Reply by mrracoonman** [reply]
Would this work if I have several? Say 20-ish? Would I have to go into the after parameter of each one to do the next one?

**Reply by lgaan** [reply]
You can have some sort of loop management system which has a callback. [Here is a pretty good playlist example gist](https://gist.github.com/EvieePy/ab667b74e9758433b3eb806c53a19f34)
